### DISCUSSION TITLE
Fileserver::from(relative!(path)) causes runtime to fail on compile time path

### DISCUSSION THREAD

**Discussion by JLT032** [rwf2/Rocket#2592] [Questions]
looking at a way to server css and images with Tera templates  I started building rocket like which runs fine for the build user

```
use rocket::Build;
use rocket::fs::{FileServer, relative};
use rocket_dyn_templates::{Template, context};

#[rocket::get("/")]
fn index() -> Template {
    Template::render("mindex", context! { title: "commandline" })
}

#[rocket::launch]
fn rocket() -> rocket::Rocket<Build> {
    rocket::build().attach(Template::fairing())
            .mount("/", rocket::routes![index])
            .mount("/css", FileServer::from(relative!("css")))
            .mount("/img", FileServer::from(relative!("img")))
}

```
the compiled binary runs fine from the build directory

However, when copying the binary to different directory there is an alert stating the 

     Error: FileServer path '<srcode-build-path-here>/css is not a directory.
     >> Aborting early to prevent inevitable handler failure.
    thread 'main' panicked at 'invalid directory: refusing to continue', src/main.rs:15:28
    note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

I'm very new to both rust and rocket and i don't know if there's a build flag required to build binaries in non-absolute way so the `<srcode-build-path-here>` is omitted.

The binary works fine for the user used to 'cargo build --release' the binary but throws the for mentioned error when ran for a different user. I'm unable to find related documentation.

Thanks for your assist

**Comment by JLT032** [comment]
does work when writing

      .mount("/css", FileServer::from(relative!("css")))
      .mount("/img", FileServer::from(relative!("img")))
as
      .mount("/css", FileServer::from("/full/destination/path/here/css"))
      .mount("/img", FileServer::from("/full/destination/path/here/img"))
     
for now leaving this as a potential bug report

**Comment by SergioBenitez** [comment]
The [docs for `relative!()`](https://api.rocket.rs/v0.5-rc/rocket/fs/macro.relative.html) should elucidate why this is happening. The (absolute) path is computed at compile-time. If you move the source directory, or move the binary to another machine, then the path resolved at compile-time will no longer exist. If you just want a relative path, you don't need to use `relative!()`. Just use the path directly, like `FileServer::from("css")`. That will require there to be a `css` directory in the CWD at the time the binary is run.
